ktor-utils/common/src/io/ktor/util/converters/DataConversion.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.converters
+
+import io.ktor.util.*
+import io.ktor.util.reflect.*
+import kotlin.reflect.*
+
+/**
+ * Data conversion feature to serialize and deserialize types using [converters] registry
+ */
+public class DataConversion(configuration: Configuration) : ConversionService {
+    private val converters: Map<KClass<*>, ConversionService> = configuration.converters.toMap()
+
+    override fun fromValues(values: List<String>, type: TypeInfo): Any? {
+        val converter = converters[type.type] ?: DefaultConversionService
+        return converter.fromValues(values, type)
+    }
+
+    override fun toValues(value: Any?): List<String> {
+        val type: KClass<*> = value?.let { it::class } ?: return listOf()
+        val converter = converters[type] ?: DefaultConversionService
+        return converter.toValues(value)
+    }
+
+    /**
+     * Data conversion service configuration
+     */
+    public class Configuration {
+        internal val converters = mutableMapOf<KClass<*>, ConversionService>()
```

Comment:
Please add native mutation tests as well

Reply:
There is test `testInstalledConversion` that mutates this map in the `common` test module. Is there anything else to test?

Reply:
thanks, they are good